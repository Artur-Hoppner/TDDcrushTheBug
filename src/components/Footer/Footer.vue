!<template>
  <div>
    <footer>
      <p>Copyright â“’ {{ new Date().getFullYear() }}</p>
      <p>Heisenberg enterprise</p>
    </footer>
  </div>
</template>

<script>
export default {};
</script>

<style lang="scss" scoped>
@import url('https://fonts.googleapis.com/css2?family=Montserrat:ital@1&family=Raleway&display=swap%27');

footer {
  font-family: 'Raleway', sans-serif;
  background: #a5d1bc;
  padding: 1%;
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
}
</style>
