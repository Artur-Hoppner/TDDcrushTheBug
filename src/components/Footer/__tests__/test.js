import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Footer from '@/components/Footer/Footer.vue';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Test footer rendering and elements', () => {
  test('check that component is rendered', () => {
    //Arrange
    const wrapper = shallowMount(Footer, { localVue });
    //Assert
    expect(wrapper.element).toMatchSnapshot();
  });

  test('Check if static element values exists', () => {
    //Arrange
    const wrapper = shallowMount(Footer, { localVue });
    const timeStamp = new Date().getFullYear();
    //Act
    const paragraph = wrapper.find('p');
    const textContent = wrapper.text();
    //Assert
    expect(paragraph.exists()).toBe(true);
    expect(textContent).toBe('Copyright â“’ ' + timeStamp);
  });
});
