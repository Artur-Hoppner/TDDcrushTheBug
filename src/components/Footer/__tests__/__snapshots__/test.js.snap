// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test footer rendering and elements check that component is rendered 1`] = `
<div>
  <footer>
    <p>
      Copyright â“’ 2020
    </p>
     
    <p>
      Heisenberg enterprise
    </p>
  </footer>
</div>
`;
